### PR TITLE
Empty states for different library category filters

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Screenshot 2023-09-05 at 4 13 16 PM
Screenshot 2023-09-05 at 4 58 38 PM

### CODE REVIEW

File: packages/common/src/store/pages/saved-page/actions.ts [hunk 1]
```diff
@@ -84,7 +83,7 @@ export const fetchSavesRequested = () => ({
   type: FETCH_SAVES_REQUESTED
 })
 
-export const fetchSavesSucceeded = (saves: Favorite[]) => ({
+export const fetchSavesSucceeded = (saves: any[]) => ({
```
Comment (amendelsohn) [inline packages/common/src/store/pages/saved-page/actions.ts:86]: Are there conflicting `Favorite` types, or are they not just favorites anymore cause of the new filters?

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

File: packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx [hunk 2]
```diff
@@ -39,6 +49,18 @@ export const AlbumsTab = () => {
...
+  let emptyTabText: string
+  if (selectedCategory === LibraryCategory.All) {
+    emptyTabText = messages.emptyAlbumAllText
+  } else if (selectedCategory === LibraryCategory.Favorite) {
+    emptyTabText = messages.emptyAlbumFavoritesText
+  } else if (selectedCategory === LibraryCategory.Repost) {
+    emptyTabText = messages.emptyAlbumRepostsText
+  } else {
+    emptyTabText = messages.emptyAlbumPurchasedText
+  }
```
Comment (amendelsohn) [inline packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx:62]: not a big deal at all, but I wish we had a better pattern for this (not just here, but we often have a message type that varies based on an enum)

maybe a `getEmptyMessageByCategory` function or map? Could even make the messages object keyed by the enum? Maybe over-engineering 😅

File: packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx [hunk 3]
```diff
@@ -25,9 +27,13 @@ import { OfflineContentBanner } from './OfflineContentBanner'
...
 
 const { getIsReachable } = reachabilitySelectors
+const { getSelectedCategory } = savedPageSelectors
 
 const messages = {
-  emptyTabText: "You haven't favorited any playlists yet.",
+  emptyPlaylistFavoritesText: "You haven't favorited any playlists yet.",
+  emptyPlaylistRepostsText: "You haven't reposted any playlists yet.",
+  emptyPlaylistAllText:
+    "You haven't favorited, reposted, or purchased any playlists yet.",
```
Comment (amendelsohn) [inline packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx:36]: is it possible to purchase playlists?

File: packages/web/src/pages/saved-page/components/emptyStateMessages.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+export const emptyStateMessages = {
```
Comment (amendelsohn) [inline packages/web/src/pages/saved-page/components/emptyStateMessages.ts:1]: could use this for mobile too?

File: packages/web/src/pages/saved-page/components/emptyStateMessages.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  emptyTrackFavoritesHeader: 'You haven’t favorited any tracks yet.',
+  emptyTrackRepostsHeader: 'You haven’t reposted any tracks yet.',
+  emptyTrackPurchasedHeader: 'You haven’t purchased any tracks yet.',
+  emptyTrackAllHeader:
+    'You haven’t favorited, reposted, or purchased any tracks yet.',
+
+  emptyAlbumFavoritesHeader: 'You haven’t favorited any albums yet.',
+  emptyAlbumRepostsHeader: 'You haven’t reposted any albums yet.',
+  emptyAlbumPurchasedHeader: 'You haven’t purchased any albums yet.',
```
Comment (amendelsohn) [inline packages/web/src/pages/saved-page/components/emptyStateMessages.ts:10]: are playlists/albums purchasable?

**Review by amendelsohn — APPROVED** [review]
Looks good!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--empty-states-library

**Pull request closed without merging**